timer.js

```diff
@@ -11,10 +11,32 @@
     // Like setTimeout, but only takes a function argument.  There's
     // no time argument (always zero) and no arguments (you have to
     // use a closure).
-    function setZeroTimeout(fn) {
+    function nextTickBeforeEvents(fn) {
         resolved.then(fn);
     }
 
     // Add the one thing we want added to the window object.
-    window.setZeroTimeout = setZeroTimeout;
+    window.nextTickBeforeEvents = nextTickBeforeEvents;
+})();
+
+(function() {
+  var cbs = [];
+  var msg = 135921;
+
+  function nextTickDuringEvents(fn) {
+    cbs.push(fn);
+    window.postMessage(msg, window.location.origin);
+  }
+
+  function recv(ev) {
+    if (window !== ev.source || ev.data !== msg) {
+      return;
+    }
+
+    cbs.shift()();
+  }
+
+  window.addEventListener("message", recv, false);
```

Comment:
> Did you make any of these changes on purpose?

I was unaware that there had ever been a postMessage-based implementation of setZeroTimeout in our repo!

I intentionally chose `window.location.origin` instead of "*" because, if we know what the origin should be, it seemed like there was no reason not to specify it, even though in our case it's not likely to give any security benefit.

Capturing and `stopPropagation` could conceivably provide a slight perf boost as well. I'll add them to this branch.